app/controllers/concerns/api/v2/lookup_keys_common_controller.rb

```diff
@@ -80,34 +67,17 @@ def smart_class_parameters_resource_scope
     params.distinct
   end
 
-  def find_smarts
-    @smarts   = @smart_variables
-    @smarts ||= @smart_class_parameters
-    @smarts
-  end
-
-  def find_smart
-    @smart   = @smart_variable
-    @smart ||= @smart_class_parameter
-    @smart
-  end
-
   def return_if_smart_mismatch
-    if (@smarts && @smart && !@smarts.find_by_id(@smart.id)) || (@smarts && !@smart)
-      obj = smart_variable_id? ? "Smart variable" : "Smart class parameter"
-      id = if smart_variable_id?
-             params.key?('smart_variable_id') ? params['smart_variable_id'] : params['id']
-           else
-             params.key?('smart_class_parameter_id') ? params['smart_variable_id'] : params['id']
```

Comment:
I think this check is still needed, depending on if it's the override controller or the smart class param controller. 
However, this whole callback seems to be quite complicated, perhaps it can be simplified?